### DISCUSSION TITLE
Is there a way to "include" a excel table in the qmd file.

### DISCUSSION THREAD

**Discussion by anti-destiny** [quarto-dev/quarto-cli#6373] [Feature Requests]
### Description

While we can edit tables in Markdown, spreadsheet editors like Excel/WPS/LibreOffice usually provide more powerful features. So I think it would be better if Quarto can support embedding existing spreadsheet files into the document. For example, it could be like this:

```{file}
file-format: excel
label: tbl-my-table
tbl-cap: This is an Excel table
path:  xxx/xxx/my-table.xlsx
```
or something like this: 

`{{< include _content.xlsx >}}`

Currently, a workaround used by me is to write some Python code:

```{python}
#| label: tbl-a
#| tbl-cap: ……
#| echo: false

from IPython.display import Markdown
from tabulate import tabulate
import pandas as pd 

with open('include_content.xlsx', 'rb') as f:
    tb = pd.read_excel(f)

Markdown(tabulate(
    tb, 
    tb.head(),
    tablefmt="grid"
))
```

However, merged cells or borderlines are missed due to the format conversion in the above code. So I'm expecting an official implementation of this.

Will Quarto support such features in the future? Or does anyone has a better idea/workaround for this problem? Any advice is appreciated.

**Comment by rgaiacs** [comment]
Hi @anti-destiny, thanks for your feature request. Support embedding spreadsheets is very challenge because of the many native spreadsheet features that users will expect to work out of the box, for example, cell style, variables, macros, ...

Using Python, R, or another language is the best way to support the display of spreadsheets.

**Reply by ppreshant** [reply]
That makes sense. When you say this
> Using Python, R, or another language is the best way to support the display of spreadsheets.

Did you mean some workflow such as using a code chunk that does `read_xlsx()` on the spreadsheet and uses `kable()` to render it?

**Reply by mcanouil** [reply]
Yes.
